sympy/core/relational.py

```diff
@@ -310,6 +310,12 @@ def __new__(cls, lhs, rhs=0, **options):
                 if r is not None:
                     return _sympify(r)
 
+            # If expression have both Boolean terms
+            if (isinstance(lhs, BooleanTrue) or isinstance(lhs, BooleanFalse)) and \
```

Comment:
@smichr it's looking bad but ```isinstance(lhs, Boolean)``` it's check in all ```Boolean``` class so that's make problem for checking.